apps/web/src/actions/evaluations/createFromPrompt.ts

```diff
@@ -0,0 +1,52 @@
+'use server'
+
+import {
+  EvaluationMetadataType,
+  EvaluationResultableType,
+} from '@latitude-data/core/browser'
+import { createEvaluation } from '@latitude-data/core/services/evaluations/create'
+import { z } from 'zod'
+
+import { withDocument } from '../procedures'
+
+export const createEvaluationFromPromptAction = withDocument
+  .createServerAction()
+  .input(
+    z.object({
+      name: z.string(),
+      prompt: z.string(),
+      type: z.union([z.literal('number'), z.literal('boolean')]),
+      metadata: z
+        .object({
+          range: z.object({
+            from: z.number(),
+            to: z.number(),
+          }),
+        })
+        .optional(),
+    }),
+    { type: 'json' },
+  )
+  .handler(async ({ input, ctx }) => {
+    const result = await createEvaluation({
+      workspace: ctx.workspace,
+      name: input.name,
+      description: 'AI-generated evaluation',
+      type: EvaluationMetadataType.LlmAsJudge,
+      configuration: {
+        type:
+          input.type === 'number'
+            ? EvaluationResultableType.Number
+            : EvaluationResultableType.Boolean,
+        detail: input.type === 'number' ? input.metadata : undefined,
```

Comment:
All this could be validated inside zod so this checks are not needed. `refine` is what you're looking for

Reply:
```javascript
const allowedTypesSchema = z.enum([
  EvaluationResultableType.Number,
  EvaluationResultableType.Boolean,
]);

const inputSchema = z
  .object({
    name: z.string(),
    prompt: z.string(),
    type: allowedTypesSchema, // Using the restricted enum values
    metadata: z
      .object({
        range: z.object({
          from: z.number(),
          to: z.number(),
        }),
      })
      .optional(),
  })
  .refine((data) => {
    // If type is 'Number', metadata must be present
    if (data.type === EvaluationResultableType.Number && !data.metadata) {
      return false;
    }
    return true;
  }, {
    message: "Metadata is required when type is 'Number'",
    path: ['metadata'],
  })
```

Reply:
Also instead of metadata call it `configDetail`

Reply:
will do in another PR